src/Context/LoginContext.php

```diff
@@ -245,22 +266,48 @@ public function stepPasswordForEmailShouldBe($id, $password)
      * @return Member
      */
     protected function generateMemberWithPermission($email, $password, $permCode)
+    {
+        $group = $this->generateGroupWithPermission($permCode);
+
+        return $this->generateMember($email, $password, $group, $permCode);
+    }
+
+    /**
+     * Get or generate a group with the given permission code
+     *
+     * @param string $groupName
+     * @return Member
+     */
+    protected function generateGroupWithPermission($permCode)
     {
         // Get or create group
-        $group = Group::get()->filter('Title', "$permCode group")->first();
+        $group = Group::get()->filter('Title', "$permCode")->first();
         if (!$group) {
             $group = Group::create();
+            $group->Title = "$permCode";
```

Comment:
```suggestion
        $group = Group::get()->filter('Title', "$permCode group")->first();
        if (!$group) {
            $group = Group::create();
            $group->Title = "$permCode group";
```
We don't want to change the behaviour of generateMemberWithPermission() - that can still add "group" to the end of the group name.